targets/TARGET_STM/i2c_api.c

```diff
@@ -261,6 +261,7 @@ void i2c_sw_reset(i2c_t *obj)
 void i2c_init(i2c_t *obj, PinName sda, PinName scl)
 {
 
+    memset(obj, 0, sizeof(*obj));
```

Comment:
sorry but i2c_init is also used internally in several places in this file in order to reset the IP in case of communication issues. When resetting the IP, we don't want to lose the current frequency. 
That's why we actually check the value of obj_s->hz

Reply:
Ok yes good catch, I'll post an alternate fix

Reply:
updated PR